src/pages/iou/request/step/IOURequestStepConfirmation.tsx

```diff
@@ -201,25 +203,32 @@ function IOURequestStepConfirmation({
 
     // When the component mounts, if there is a receipt, see if the image can be read from the disk. If not, redirect the user to the starting step of the flow.
     // This is because until the request is saved, the receipt file is only stored in the browsers memory as a blob:// and if the browser is refreshed, then
-    // the image ceases to exist. The best way for the user to recover from this is to start over from the start of the request process.
+    // the image ceases to exist. We will navigate to the user to the scanning page and resume the flow once the user has re-uploaded the file
     // skip this in case user is moving the transaction as the receipt path will be valid in that case
-    useEffect(() => {
-        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-        const isLocalFile = FileUtils.isLocalFile(receiptPath);
+    useFocusEffect(
+        useCallback(() => {
+            const backToParam = ROUTES.MONEY_REQUEST_STEP_CONFIRMATION.getRoute(action, iouType, transactionID, reportID);
+            if (!receiptPath && requestType === CONST.IOU.REQUEST_TYPE.SCAN) {
+                Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_SCAN.getRoute(CONST.IOU.ACTION.CREATE, iouType, transactionID, reportID, backToParam));
+                return;
+            }
```

Comment:
Why is this change required or what bug is it fixing?

Reply:
Ah I see. Nice catch. For that we should just change the `backTo` to the start page. We'd also need to pass `shouldEnforceFallback` as true in the receipt scan page.

Reply:
- After user refresh the page and is navigated to the scan page, clicking the back button at this point should go back to the start page. Other pages (e.g. participant selection page) does not make sense because we won't have a receipt.
- If you can make it work without `shouldEnforceFallback=true` that's totally fine. I just didn't get this working as the fallback route was not in stack

Reply:
Navigating to the start page should be the simplest solution. You just need to modify navigateToStartStepIfScanFileCannotBeRead.onFailure callback adding an early return:
```js
        if (requestType === CONST.IOU.REQUEST_TYPE.SCAN) {
            Navigation.navigate(
                ROUTES.MONEY_REQUEST_STEP_SCAN.getRoute(CONST.IOU.ACTION.CREATE, iouType, transactionID, reportID, ROUTES.MONEY_REQUEST_START.getRoute(iouType, requestType)),
            );
            return;
        }
```

Then you can revert most of the other changes e.g. use of `useFocusEffect` and the extra navigation

Reply:
The `backTo` param won't be used in that case because it won't exist in the callstack. For the back button though we want to enforce it with `shouldEnforceFallback=true`



https://github.com/Expensify/App/assets/16493223/af664838-a4f1-42a3-a62c-881bc7dc9bc1

Reply:
@dominictb I'm trying to avoid "chasing the user" from page to another and instead just make them land in the correct page. Maybe we are missing a simpler solution here?

> It's already navigated to the scan step. Just that they navigate back using route.params.reportID instead of transaction.reportID -> causing the the money request to be re-init. If we want to use the transaction.reportID to nav back in this case, we'll (accidentally, or intentionally) force user to re-select participants.

Perhaps fixing the user re-selection would be more straightforward?

Reply:
I have also tried the solution in this PR and I'm getting stuck in the scan page


https://github.com/Expensify/App/assets/16493223/75abb52d-9c34-450e-a8c8-75c6b3f704c6

Reply:
@dominictb Redirecting the user to the start sounds good to me then. Can you please implement that?

Reply:
@dominictb Unfortunately this generated too much change. Let's just overwrite the go back functionality in the scan page if the request type is scan. Testing this and looks we only need the following changes:

`IOU.navigateToStartStepIfScanFileCannotBeRead`
```diff
-        if (requestType === CONST.IOU.REQUEST_TYPE.MANUAL) {
+        if (requestType === CONST.IOU.REQUEST_TYPE.MANUAL || requestType === CONST.IOU.REQUEST_TYPE.SCAN) {
```

`IOURequestStartPage.resetIOUTypeIfChanged`
```diff
+            if (transaction?.iouRequestType === newIOUType) {
+                return;
+            }
```

`IOURequestStepConfirmation.navigateBack`
```diff
-            Navigation.goBack(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, reportID, undefined, action));
+            Navigation.goBack(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, transaction?.reportID || reportID, undefined, action));
```

`IOURequestStepScan.navigateBack`
```diff
+        if (transaction?.iouRequestType === CONST.IOU.REQUEST_TYPE.SCAN) {
+            Navigation.goBack(ROUTES.MONEY_REQUEST_START.getRoute(iouType, transaction?.iouRequestType), true);
+            return;
+        }
```

Reply:
That looks like a workaround and it's hard to reason about. `initMoneyRequest` should reinit all the fields

Reply:
My bad, I didn't read the solution well, I thought we are comparing the request type with `scan` type. Comparing the request type with the previous one looks good to me. Let's do that and test it